mautic/mautic
# Simplifying grunt

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [shouldn't be]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes https://mautic.atlassian.net/browse/TPROD-465



#### Description:

~~⚠️ this PR depends on https://github.com/mautic/mautic/pull/12373 to avoid conflicts. Merge this PR afterwards.~~

While looking into https://mautic.atlassian.net/browse/TPROD-465 I noticed that the `grunt-contrib-cssmin` module is not configured in `Gruntfile.js`. I tried to remove it and run the `grunt compile-less` command and there was no change in the generated css files nor any error. I went through the other grunt packages and found that the same applies for `grunt-autoprefixer` and `grunt-contrib-copy`. I removed those as they weren't configured.

I also removed `load-grunt-tasks` as it was saving just just one line as we use 2 Grunt modules now and so it's not a big problem to list both of them instead of autoloading them.

And I also removed `time-grunt` as it's just showing time in the output. It's unnecessary and it also resolved some vulnerability warnings so it was well worth it.

From:
```
14 vulnerabilities (1 low, 2 moderate, 6 high, 5 critical)
```
to:
```
10 vulnerabilities (1 low, 4 high, 5 critical)
```

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Run `grunt compile-less`
3. Make some change in some `.less` file. For example in https://github.com/mautic/mautic/blob/5.x/app/bundles/CoreBundle/Assets/css/app/less/custom.less add `body{background:red}` statement and watch that the css files were updated with this change. Remove it and the autogenerated files will get back to its original form.
